### PR TITLE
Better types

### PR DESCRIPTION
**Pull request by Aqours**

### CODE REVIEW

File: d.ts/flv.d.ts [hunk 1]
```diff
@@ -141,39 +244,73 @@ declare namespace FlvJs {
...
+        url: string;
+        decodedFrames?: number;
+        droppedFrames?: number;
+    }
+
+    interface FlvPlayerStatisticsInfo {
+        playerType: 'FlvPlayer';
+        decodedFrames?: number;
+        droppedFrames?: number;
+    }
```
Comment (xqq) [inline d.ts/flv.d.ts]: There're more fields in FlvPlayer's StatisticsInfo
https://github.com/Bilibili/flv.js/blob/master/src/core/transmuxing-controller.js#L418

File: d.ts/flv.d.ts [hunk 2]
```diff
@@ -141,39 +244,89 @@ declare namespace FlvJs {
...
 
-    /**
-     * @deprecated Use FlvJs.Player instead.
-     */
-    // tslint:disable-next-line
-    interface NativePlayer extends Player {}
+    interface FlvPlayerReportStatisticsInfo {
+        url: string;
+        hasRedirect: boolean;
+        redirectedURL?: string;
```
Comment (xqq) [inline d.ts/flv.d.ts]: `redirectedURL` is an optional field which exists only if `hasRedirect === true`.
`redirectedURL?: string | undefined` may be better?

Comment (Aqours) [reply d.ts/flv.d.ts]: I think `redirectedURL?: string`  equal to `redirectedURL: string | undefined`.  Optional field indicate that `redirectedURL` maybe `undefined`.

Comment (xqq) [reply d.ts/flv.d.ts]: Oh I didn't notice the `?`

**Comment by xqq** [comment]
Merged,
thanks!

**Pull request merged**